src/build/auto-compile-helper.ts

```diff
@@ -44,8 +54,12 @@ export const generateAutoCompileCode = (
         folder = File(myComputer, currentFolderPath)
         if folder and folder.get_files.len == 0 and folder.get_folders.len == 0 then
           push(remainingFolderPaths, path(parent(folder)))
-          delete(folder)
-          print("Deleted " + folder.path)
+          
+          if deleteAllImportedFoldersAfterAutoCompilation == true then
+            delete(folder)
+            
+            print("Deleted " + folder.path)
+          end if
```

Comment:
This will make the default method of deleting left over folders not work anymore. You probably would want to do the following:
```ts
if folder and ((folder.get_files.len == 0 and folder.get_folders.len == 0) or deleteAllImportedFoldersAfterAutoCompilation) then
          push(remainingFolderPaths, path(parent(folder)))
          delete(folder)
          print("Deleted " + folder.path)
end if
```
Since the intention of the purge flag is to delete the folder regardless of their content. The default method will only delete folders if those are empty.

Reply:
As shown in the following series of messages logged by greybel-js to the terminal:
```
Creating installer.
Importing files ingame.
Trying to connect to 127.0.0.1:7777 (attempt: 0)
Imported /test-gs-prg.src successful
Imported /libs/math-lib.src successful
Build done in /usr/bin
Deleted /usr/bin/test-gs-prg.src
Deleted /usr/bin/libs/math-lib.src
Deleted /usr/bin/libs
2 files got imported to /usr/bin!
```
In this case, the following files and folders are being imported into Grey Hack:
```
- test-gs-prg.src
- libs
 - math-lib.src
```
As seen in the messages logged by greybel-js, the ``test-gs-prg.src`` and ``math-lib.src`` files are deleted first before the ``libs`` folder is even iterated over.

As a result, one other thing I was curious about was what is the purpose of the following check?
```
folder.get_files.len == 0 and folder.get_folders.len == 0
```
In the following line of code:
```
if folder and ((folder.get_files.len == 0 and folder.get_folders.len == 0) or deleteAllImportedFoldersAfterAutoCompilation) then
```

Reply:
Anyways, I have implemented the change that you suggested and have pushed the relevant changes to my fork of this repository.

However, I would still appreciate it if you could get back to me regarding the 2 questions that I asked in my previous comments.

Reply:
Thanks for implementing the change.

In regards of your questions:
So let's go over the two behaviours we want to have available:
1.) The first one is the default behaviour. In the default behaviour we want files and folders to be cleaned up after building them. But we don't want to be aggressive about deleting folders which means if there are files which are unrelated to the building process in the folder the folder should remain. We archive this via this condition:  `folder.get_files.len == 0 and folder.get_folders.len == 0`. By using this condition we make sure that we only delete if there is nothing remaining in the folder. Let's take two examples into consideration.

Scenario 1#:
My in-game file system:
```
- home
 - guest
 - user
  - Config
   - Bank.txt
```

My project:
```
- test-gs-prg.src
- libs
 - math-lib.src
```
In this scenario we just simply import the project into `/home/user`. The default behaviour will create all files and folders, build and then delete everything again since there is no pre existing folders or files which overlap in anyway with any of the project folders.

Scenario 2#:
My in-game file system:
```
- home
 - guest
 - user
  - libs
   - my-test-file.txt
  - Config
   - Bank.txt
```

My project:
```
- test-gs-prg.src
- libs
 - math-lib.src
```
In this scenario we import the project into `/home/user` again. The default behaviour will create all files and folders, build and then delete every file but keep the  `/libs` folder. The reason why the `/libs` folder is kept is because there is an pre existing `my-test.file.txt` in  `/home/user/libs`.

2.) The second one is the purge behaviour. In the purge behaviour we want files and folders to be cleaned up after building them. We do not care if there are any pre existing files or folders. Therefore we bypass the default behaviour condition by checking if `deleteAllImportedFoldersAfterAutoCompilation` is true. Let's take a look at the two examples again.

Scenario 1#:
My in-game file system:
```
- home
 - guest
 - user
  - Config
   - Bank.txt
```

My project:
```
- test-gs-prg.src
- libs
 - math-lib.src
```
Just like default behaviour everything will be imported, build and then deleted.

Scenario 2#:
My in-game file system:
```
- home
 - guest
 - user
  - libs
   - my-test-file.txt
  - Config
   - Bank.txt
```

My project:
```
- test-gs-prg.src
- libs
 - math-lib.src
```
Unlike the default behavior for this scenario everything will be imported, build and then deleted. Due to not caring about any pre existing files or folders `my-test-file.txt` will be deleted as well.

I hope this makes it more clear.

Reply:
I see, thanks for the in-depth explanation.

I was not initially aware of the tool's intended default behavior which might have caused me to become somewhat confused regarding the code in your suggested change, however, now that you have explained it, it does indeed seem to make sense to me as well.

Thanks again for the clarification.